iceoryx_posh/include/iceoryx_posh/popo/event_callback.hpp

```diff
@@ -17,27 +17,58 @@
 #ifndef IOX_POSH_POPO_EVENT_CALLBACK_HPP
 #define IOX_POSH_POPO_EVENT_CALLBACK_HPP
 
+#include "iceoryx_utils/cxx/helplets.hpp"
+
 namespace iox
 {
 namespace popo
 {
+template <typename OriginType, typename ContextDataType>
+struct EventCallback;
+
 namespace internal
 {
 struct NoType_t
 {
 };
+
+using GenericCallbackPtr_t = void (*)();
+using GenericCallbackRef_t = void (&)();
+
+using TranslationCallbackRef_t = void (&)(void* const, void* const, GenericCallbackPtr_t const);
+using TranslationCallbackPtr_t = void (*)(void* const, void* const, GenericCallbackPtr_t const);
+
+template <typename T, typename ContextDataType>
+struct TranslateAndCallTypelessCallback
```

Comment:
`TranslateAndCallCallback`? It's not typeless anymore, but decided via the template specialisation.

Reply:
The input is typeless and this typeless input is translated into something which has types in a typesafe manner.